### ISSUE TITLE
Prevent push if `can_create?` returned false

### ISSUE THREAD

**Issue by TylerRick** [jonhue/notifications-rails#47]
I noticed that even if `can_create?` returns false, `Notification.create` still returns a `Notification` object — it's just not persisted.

But what's stopping the push from happening if notifications are completely disabled for this target user (`{enabled: false}` in their settings record) if you do something like this?:
```ruby
notification = Notification.create(target: User.first, object: Recipe.first)
notification.push(:CustomPusher, option_one: 'value_two')
```
I haven't tried it to be certain, but if I'm reading the code correctly, this would _still_ push (deliver) the notification — even though it couldn't be "created".

In other words, it doesn't appear that being "_created_" (saved/persisted) is a requisite state for a push to be allowed — the object need only be *initialized*.

Should there be an additional check for `can_create?` inside of `can_push?`? Or is it up to the end user to handle this, something like:
```ruby
notification = Notification.create(target: User.first, object: Recipe.first)
if notification.can_create?
  notification.push(:CustomPusher, option_one: 'value_two')
end
```
... except that `can_create?` is currently marked `private` so you can't do that. Should we make it public, to allow for such checks as this?

**Comment by jonhue** [comment]
Good spot! This should definitely be handled in `can_push?`.
